### PR TITLE
Refactor protocol adapters to support non spring based implementations

### PR DESCRIPTION
**Pull request by dejanb**
This PR refactors a bit adapters maven modules to better support non Spring based implementations. The actual changes are:
* Remove Spring dependencies from the parent module and introduce them only to the actual adapter implementation
* Move Vert.x implementation classes to the base modules, so that they can be reused

### CODE REVIEW

File: adapters/base-spring/pom.xml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        <dependency>
+          <groupId>org.springframework.boot</groupId>
+          <artifactId>spring-boot-autoconfigure</artifactId>
+        </dependency>
+        <dependency>
+          <groupId>org.springframework.boot</groupId>
+          <artifactId>spring-boot-starter-logging</artifactId>
+        </dependency>
+    </dependencies>
+
```
Comment (ctron) [inline adapters/base-spring/pom.xml:31]: Double empty line :grin:

Comment (dejanb) [reply adapters/base-spring/pom.xml:31]: Fixed

File: adapters/base-spring/pom.xml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+</project>
```
Comment (sophokles73) [inline adapters/base-spring/pom.xml:32]: missing EOL

**Comment by dejanb** [comment]
> why do you want to move the CoAP adapter implementation class?

@sophokles73 I moved all vert.x based implementation to the base modules (where they exists), so that they can be reused between Spring and Quarkus deployments. That's how HTTP adapter works for now, but I can imagine we do the same for MQTT, CoAP, etc.

**Pull request merged**